src/frontend/src/optimizer/plan_node/stream_changelog.rs

```diff
@@ -21,20 +21,23 @@ use super::stream::prelude::PhysicalPlanRef;
 use super::utils::impl_distill_by_unit;
 use super::{ExprRewritable, PlanBase, PlanTreeNodeUnary, Stream, StreamNode, generic};
 use crate::optimizer::StreamPlanRef as PlanRef;
-use crate::optimizer::property::{MonotonicityMap, StreamKind};
+use crate::optimizer::property::{Distribution, MonotonicityMap, StreamKind};
 use crate::stream_fragmenter::BuildFragmentGraphState;
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub struct StreamChangeLog {
     pub base: PlanBase<Stream>,
     core: generic::ChangeLog<PlanRef>,
+    distribution_keys: Vec<u32>,
 }
 
 impl StreamChangeLog {
-    pub fn new(core: generic::ChangeLog<PlanRef>) -> Self {
+    pub fn new_with_dist(
+        core: generic::ChangeLog<PlanRef>,
+        dist: Distribution,
+        distribution_keys: Vec<u32>,
```

Comment:
I think we can obtain this from `core.input.distribution().dist_column_indices()`